### PR TITLE
when getNextRelease from koji fails, find next release

### PR DESCRIPTION
**Pull request by rcerven**
* OSBS-8652

Signed-off-by: Robert Cerven <rcerven@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [ ] Pull request includes link to an osbs-docs PR for user documentation updates
- [ ] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/pre_bump_release.py [hunk 1]
```diff
@@ -102,7 +102,10 @@ def next_release_general(self, component, version, release, release_label,
     def get_next_release_standard(self, component, version):
         build_info = {'name': component, 'version': version}
         self.log.debug('getting next release from build info: %s', build_info)
-        next_release = self.get_patched_release(self.xmlrpc.getNextRelease(build_info))
+        try:
+            next_release = self.get_patched_release(self.xmlrpc.getNextRelease(build_info))
+        except koji.BuildError:
```
Comment (athos-ribeiro) [inline atomic_reactor/plugins/pre_bump_release.py]: Could we add a comment on why we would reach a koji buildError here?

**Comment by athos-ribeiro** [comment]
LGTM

It would be nice to add more explanation to the reasoning behind the patch. For instance, we could add some description of how isolated builds could be an issue in the commit message

File: atomic_reactor/plugins/pre_bump_release.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_bump_release.py]: added

**Comment by rcerven** [comment]
added explanation to the commit message

File: atomic_reactor/plugins/pre_bump_release.py [hunk 2]
```diff
@@ -138,6 +143,21 @@ def get_next_release_append(self, component, version, base_release, base_suffix=
                     return next_release
             suffix += 1
 
+    def get_next_release(self, build_info):
+        queryopts = {'order': '-build.id', 'limit': 1}
+        # release is either single number or with decimal point, so we can easily bump next release
+        # pylint: disable=W1401
+        search_str = '^{}-{}-\d+(\.\d+)?$'.format(build_info['name'], build_info['version'])
```
Comment (chmeliik) [inline atomic_reactor/plugins/pre_bump_release.py]: Is there a reason not to use an r-string here? That should get rid of the pylint warning.

Comment (rcerven) [reply atomic_reactor/plugins/pre_bump_release.py]: added 'r'

**Comment by ktdreyer** [comment]
If we update koji's `getNextRelease` to incorporate the changes that you have here, can we drop this from OSBS?

**Pull request merged**